ansible/deploy_db.yml

```diff
@@ -99,3 +99,43 @@
     - celery
   roles:
     - {role: java, tags: java}
+
+# Disables transparent huge pages (THP)
+- name: Disable THP
+  become: true
+  hosts:
+    - postgresql
+    - couchdb
+    - redis
+    - elasticsearch
+    - rabbitmq
+    - zookeeper
+    - riakcs
+    - formplayer
+  tags: 'kernel_tuning'
+  tasks:
+    - stat:
+        path: /sys/kernel/mm/transparent_hugepage/enabled
+      register: thp_enabled
+    - stat:
+        path: /sys/kernel/mm/transparent_hugepage/defrag
+      register: thp_defrag
+    - name: Sets THP enabled to never
+      raw: echo never > /sys/kernel/mm/transparent_hugepage/enabled
+      when: thp_enabled.stat.exists
+    - name: Sets THP defrag to never
+      raw: echo never > /sys/kernel/mm/transparent_hugepage/defrag
+      when: thp_defrag.stat.exists
+    - name: Adds transparent_hugepage=never to default grub config
```

Comment:
just for our curren peace of mind, i tried:
grub-mkconfig > /tmp/grub.tmp
diff /tmp/grub.tmp /boot/grub/grub.cfg
with no difference, but as I said I think it's better to leave grub.cfg alone and let grub re-configure it with the parameter in /etc/default/grub which syntax is a lot easier to check for humans.